src/java/org/apache/cassandra/transport/Message.java

```diff
@@ -200,7 +206,9 @@ public String debugString()
 
     public static abstract class Request extends Message
     {
+        private volatile boolean isExecutingAsync = false;
         private boolean tracingRequested;
+        private final long creationTimeNanos = MonotonicClock.approxTime.now();
```

Comment:
AFAIU, this patch relies on the assumption this is populated as early as possible, namely when the message is read from the network.

Unfortunately, on the CNDB side this is not the case courtesy of Stargate, as here's where request messages are created: https://github.com/riptano/cndb/blob/25001cb6a8301d1f70bcd18f36db1ea9add0ca71/cndb-stargate/cndb-persistence/src/main/java/io/stargate/db/dse/impl/CndbPersistence.java#L666

AFAIU, that means we would "start measuring" later, and after some possibly time consuming actions such as auth (via OPA).

Am I making sense? Is that something you considered?

Reply:
I think this change is useful as is. We can look at making it better by moving the timestamp creation even earlier in a follow on improvement if we think that is significant.

Reply:
It is good, but it might not cover a range of incidents such as the recent OPA one which we told DBPE would be solved by this patch. As long as we agree about that and make everyone aware, I'm fine.

Reply:
Ignore me here. As also verified by @jkni, auth calls happen in the `StargateQueryHandler` after the message created here is enqueued on the NTR, so all good. Apologies I should have verified this upfront.